worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupHeadersAndBody.java

```diff
@@ -166,7 +181,15 @@ else if (line.contains(": ")) {
                 else if (line.contains(":*")) {
                     bodyIndex++;
                     addStandardisedHeader(nattyParser, headersElement, lines, line, ":\\*");
-                }else {
+                } // Only enter this block if we get a match i.e. line is "On xxx, abc wrote:"
+                else if (containsWhatSearched(line)) {
+                    if (matcher.find()) {
+                        bodyIndex++;
+                        addCondensedHeader(nattyParser, headersElement, line, matcher);
+                    } else {
+                        break;
+                    }
+                } else {
```

Comment:
If this is correct then you should use `&&` so that you don't have to repeat the `break;`, but like I said above I want to discuss this getting moved:

    } // Only enter this block if we get a match i.e. line is "On xxx, abc wrote:"
    else if (containsWhatSearched(line) && matcher.find()) {
        bodyIndex++;
        addCondensedHeader(nattyParser, headersElement, line, matcher);
    } else {

Reply:
I have not joined the find with the method call, because I wanted to reduce the calls to matcher.find(). Since the pattern is very complex, a call with parallelstream(), anyMatch() and contains() could potentially save some time.